pkgs/by-name/yt/yt-dlp/package.nix

```
@@ -1,46 +1,37 @@
 { lib
-, buildPythonPackage
-, fetchPypi
-, brotli
-, hatchling
-, certifi
-, ffmpeg
+, python3Packages
+, ffmpeg-headless
 , rtmpdump
 , atomicparsley
-, pycryptodomex
-, websockets
-, mutagen
-, requests
-, secretstorage
-, urllib3
 , atomicparsleySupport ? true
 , ffmpegSupport ? true
 , rtmpSupport ? true
 , withAlias ? false # Provides bin/youtube-dl for backcompat
 , update-python-libraries
 }:
 
-buildPythonPackage rec {
+python3Packages.buildPythonApplication rec {
   pname = "yt-dlp";
   # The websites yt-dlp deals with are a very moving target. That means that
   # downloads break constantly. Because of that, updates should always be backported
   # to the latest stable release.
   version = "2024.7.16";
   pyproject = true;
 
-  src = fetchPypi {
+  src = python3Packages.fetchPypi {
```

Afaicr we use top-level fetcher:

```suggestion
  src = fetchPypi {
```